nixos/modules/programs/regreet.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+let
+  cfg = config.programs.regreet;
+  settingsType = pkgs.formats.toml { };
+in
+{
+  options.programs.regreet = {
+    enable = lib.mkEnableOption (lib.mdDoc "ReGreet, a greetd greeter");
+
+    package = lib.mkPackageOptionMD pkgs "regreet" { };
+
+    settings = lib.mkOption {
+      inherit (settingsType) type;
+      default = { };
+      description = lib.mdDoc ''
+        ReGreet configuration file. Refer
+        <https://github.com/rharish101/ReGreet/blob/main/regreet.sample.toml>
+        for options.
+      '';
+    };
+
+    extraCss = lib.mkOption {
+      type = lib.types.either lib.types.path lib.types.lines;
+      default = "";
+      description = lib.mdDoc ''
+        Extra CSS rules to apply on top of the GTK theme. Refer to
+        [GTK CSS Properties](https://docs.gtk.org/gtk4/css-properties.html) for
+        modifiable properties.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.greetd.enable = true;
+
+    environment.etc = {
+      "greetd/regreet.toml".source = settingsType.generate "regreet.toml" cfg.settings;
+      "greetd/regreet.css".source = pkgs.writeText "regreet.css" cfg.extraCss;
```

Comment:
Does `pkgs.writeText` allow for a `lib.types.path` to be passed as a second parameter? I was expecting to see this line changed to
```nix
{
  "greetd/regreet.css" =
    if lib.isPath cfg.extraCss
    then {source = cfg.extraCss;}
    else {text = cfg.extraCss;};
}
```

Reply:
It is possible that users may also want to do something similar for `regreet.toml`, however I would consider this "discouraged" since there is a `pkgs.formats.toml` to use. CSS is a different story for obvious reasons.

Reply:
Sure, I'll try to modify the functionality of the toml settings as well.



> Does `pkgs.writeText` allow for a `lib.types.path` to be passed as a second parameter? I was expecting to see this line changed to

I forgot :sweat_smile: